istioctl/pkg/gencert/genCert.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package gencert
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"istio.io/istio/security/pkg/pki/ca"
+	"istio.io/istio/security/pkg/pki/util"
+)
+
+const (
+	// caCertID is the CA certificate chain file.
+	caCertID = "ca-cert.pem"
+	// caPrivateKeyID is the private key file of CA.
+	caPrivateKeyID = "ca-key.pem"
+	timeLayout     = "Jan 2 15:04:05 2006"
+)
+
+// GenerateCertKayAndExtractRootCert will generate key and certificate for workloads using citadel as CA and extract root cert out
+func GenerateCertKayAndExtractRootCert(opts util.CertOptions, client kubernetes.Interface, ns string) ([]byte, []byte, []byte, error) {
+	caSecret, scrtErr := client.CoreV1().Secrets(ns).Get(ca.CASecret, metav1.GetOptions{})
```

Comment:
Sorry, you are still directly accessing the root cert. This isn't a good practice. We need to be careful about this. Can you investigate my suggestion:
https://github.com/istio/istio/issues/21990#issuecomment-607509848 ?